src/file_pipe_log/pipe.rs

```diff
@@ -199,7 +216,14 @@ impl<F: FileSystem> SinglePipe<F> {
 
         writable_file.writer.close()?;
 
-        let (path_id, handle) = self.new_file(new_seq)?;
+        let (path_id, handle) = {
+            if let Some((id, path)) = self.recycle_file(new_seq) {
```

Comment:
Why not open it inside too? So the code can be more concise like `self.recycle_file(seq).unwrap_or(|| self.new_file(seq))?`

Reply:
Several things for the reason why I coded as this format:
- If we do just like u recommend, just like:
```Rust
let (path_id, handle) = self
            .recycle_file(new_seq)
            .unwrap_or_else(|_| self.new_file(new_seq).unwrap());
```
It will `panic` if `new_file` returns a `Err`.

- If we want to return the `Err` to above, just like:
```Rust
let (path_id, handle) = self
            .recycle_file(new_seq)
            .unwrap_or_else(|_| self.new_file(new_seq)?);
```
It's invalid in rust syntax as:
```Rust
error[E0277]: the `?` operator can only be used in a closure that returns `Result` or `Option` (or another type that implements `FromResidual`)
   --> src/file_pipe_log/pipe.rs:226:55
    |
226 |             .unwrap_or_else(|_| self.new_file(new_seq)?);
    |                             --------------------------^
    |                             |                         |
    |                             |                         cannot use the `?` operator in a closure that returns `(usize, <F as FileSystem>::Handle)`
    |                             this function should return `Result` or `Option` to accept `?`
    |
    = help: the trait `FromResidual<std::result::Result<Infallible, errors::Error>>` is not implemented for `(usize, <F as FileSystem>::Handle)`
```

- Also, we can code it just like:
```Rust
let (path_id, handle) = self
            .recycle_file(new_seq)
            .unwrap_or(self.new_file(new_seq)?);
```
But this is not what we expect, as the `unwrap_or(function(...))` format will try to get the value of `function(...)` and pass it to `unwrap_or` directly:
```
Arguments passed to unwrap_or are eagerly evaluated;
```

Reply:
By opening inside, the recycle_file will return a Option<Result<File>>, and new_file returns Result<File>, you can fuse them with recycle_file().unwrap_or_else(|| self.new_file())?